File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(br.readLine());

        while (t-- > 0) {
            String input = br.readLine();
            String[] inputArray = input.split("\\s+");
            ArrayList<Integer> a = new ArrayList<>();

            for (String s : inputArray) {
                a.add(Integer.parseInt(s));
            }

            Solution ob = new Solution();
            ob.sort012(a);

            for (int num : a) {
                System.out.print(num + " ");
            }
            System.out.println();
        }
    }
}

// } Driver Code Ends



class Solution {
    public void sort012(ArrayList<Integer> arr) {
        int n=arr.size();
        int start=0;
        int mid=0;
        int end=n-1;
        while(mid<=end){
            if(arr.get(mid)==0){
                Collections.swap(arr, start, mid);
                start++;
                mid++;
            }else if(arr.get(mid)==1){
                mid++;
            }else {
                Collections.swap(arr, mid, end);
                end--;
            }
        }
    }
}

File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void sort012(vector<int>& arr) {
        
        int low=0;
        int high=arr.size()-1;
        int mid=0;
        while(mid<=high){
            int ele=arr[mid];
            if(ele==0){
                swap(arr[mid], arr[low]);
                mid++;
                low++;
            }
            else if(ele==1){
                mid++;
            }
            else{
                swap(arr[mid], arr[high]);
                high--;
                
            }
        }
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
        cout << "~" << endl;
    }
    return 0;
}

// } Driver Code Ends